src/com/sap/piper/JenkinsUtils.groovy

```diff
@@ -19,6 +19,31 @@ static boolean hasTestFailures(build){
     return action && action.getFailCount() != 0
 }
 
+boolean addWarningsNGParser(String id, String name, String regex, String script, String example = ''){
+    def classLoader = this.getClass().getClassLoader()
+    def config = classLoader.loadClass('io.jenkins.plugins.analysis.warnings.groovy.ParserConfiguration', true)?.getInstance()
+
+    if(config.contains(id)){
+        return false
+    }else{
+        config.setParsers(
+            config.getParsers().plus(
+                classLoader.loadClass('io.jenkins.plugins.analysis.warnings.groovy.GroovyParser', true)?.newInstance(id, name, regex, script, example)
+            )
+        )
+        return true
+    }
+}
+
+@NonCPS
+static String getFullBuildLog(currentBuild) {
+    Reader reader = currentBuild.getRawBuild().getLogReader()
+    String logContent = org.apache.commons.io.IOUtils.toString(reader);
```

Comment:
full qualified name needed here since we have already another IOUtils ? Otherwise I would like to suggest to use import statement